pybossa/cache/site_stats.py

```diff
@@ -421,3 +421,53 @@ def management_dashboard_stats_cached():
         # reset stats for any missing stat
         map(delete_cache_group, management_dashboard_stats)
     return stats_cached
+
+def total_tasks(interval):
+    """
+    Total tasks
+    """
+    sql = text(f'''
+        SELECT count(task.id) as num_tasks
+        FROM task WHERE
+        to_timestamp(task.created, 'YYYY-MM-DD"T"HH24:MI:SS.US') > current_timestamp - interval '{interval}';
+    ''')
+    rows = session.execute(sql).fetchall()
+    return rows[0][0]
+
+def total_taskruns(interval):
+    """
+    Total taskruns
+    """
+    sql = text(f'''
+        SELECT count(task_run.id) as num_taskruns
+        FROM task_run WHERE
+        to_timestamp(task_run.finish_time, 'YYYY-MM-DD"T"HH24:MI:SS.US') > current_timestamp - interval '{interval}';
+    ''')
+    rows = session.execute(sql).fetchall()
+    return rows[0][0]
+
+def total_active_projects(interval):
+    """
+    Active projects
+    """
+    sql = text(f'''
+        SELECT count(project.id) as num_projects
+        FROM project WHERE
+        to_timestamp(project.updated, 'YYYY-MM-DD"T"HH24:MI:SS.US') > current_timestamp - interval '{interval}';
+    ''')
+    rows = session.execute(sql).fetchall()
+    return rows[0][0]
+
+def total_projects_using_component(component, interval):
+    """
+    Fetch total projects using component in the given interval
+    """
+    sql = text(f'''
+        SELECT count(project.id)
+        FROM project
+        LEFT JOIN "user" ON project.owner_id = "user".id
+        WHERE project.info->>'task_presenter' like '%{component}%' AND
```

Comment:
Does the code have adequate protection against sql injection attacks on the variable `component`? We may want to clean the value before usage to strip out non-alpha-numeric characters with a regex.

Example: Only a-z or A-Z or 0-9 between 3-10 characters:

```
^([a-z]|[A-Z]|[0-9]){3,10}$
```

https://stackoverflow.com/a/51304360/2596404

Reply:
This is handled in the request handler with the `isnumeric` [check](https://github.com/bloomberg/pybossa/blob/54f26a9409a910cb71c176dcafb94241e947bfde/pybossa/view/admin.py#L587).

Reply:
Right, but the above comment is for `component`, which would be a string?

Reply:
Oh, misread your initial comment.  'component' comes from the config (settings_local). Would we need to sanitize values provided by config?

Reply:
also use parameterized sql queries here and other places to prevent sql injection.

Reply:
+1 above comment. Also, even though the value is coming from the config for now, we should cleanse it for protection in case the values are eventually moved to come from the front-end.

Reply:
Plus all the good comments above, we will also need to cache it (one day seems a good value to me).